lib/index.js

```diff
@@ -29,6 +31,21 @@ function getSource(compilation, name) {
   return fs.readFileSync(path, { encoding: 'utf-8' });
 }
 
+/**
+ * Merges the provided objects, ensuring that the resulting object has its properties in sorted order.
+ * @template T
+ * @param {T} obj1
+ * @param {Partial<T> | undefined} obj2
+ * @returns {T}
+ */
+function mergeObjects(obj1, obj2) {
+  const mergedObj = assign({}, obj1, obj2);
+  const sortedPairs = toPairs(mergedObj).sort((e1, e2) => e1[0].localeCompare(e2[0]));
+  // @ts-ignore: 2322 The Lodash typedefs aren't smart enough to be able to tell TS that we're
+  // regenerating the object from the original key-value pairs.
+  return fromPairs(sortedPairs);
+}
```

Comment:
I followed the existing conventions in this file and used roughly ES5 JS rather than modern JS, but I'd be happy to do something different if you'd prefer. The two primary types of things are using Lodash methods instead of native built-ins (e.g. `lodash.topairs` vs. `Object.entries`) and not using destructuring in the arguments list of the `sort` comparator.